llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2430,6 +2430,30 @@ static bool isAllocSiteRemovable(Instruction *AI,
         unsigned OtherIndex = (ICI->getOperand(0) == PI) ? 1 : 0;
         if (!isNeverEqualToUnescapedAlloc(ICI->getOperand(OtherIndex), TLI, AI))
```

Comment:
I was going to say that the new checks should be in the  `isa<ConstantPointerNull>` path of isNeverEqualToUnescapedAlloc(), but on further consideration, they also apply to the other two cases (e.g. you could compare the result of two required-to-be-null aligned_allocs), so current code structure is good.